DID v2 Specification Design

ChangoBuitrago opened the pull request:
**Description**

This task focuses on designing the core structure of Hedera DID Specification v2. The objective is to align the new model with the W3C DID Core standard like controller delegation and message authentication through a proof object. It involves redesigning the DID Document schema, verification relationships, and DID event structures (e.g., create, update, revoke). The output of this task will serve as the technical foundation for the rest of the specification and SDK development.

**Acceptance Criteria**
- Key components from W3C DID Core are reviewed and understood
- A new DID Document schema is defined, including proof, and controller
- Updates to verification relationships are clearly documented
- Event model changes (create, update, revoke, deactivate) are designed and mapped
- Dependencies between new document fields and resolver/verifier logic are outlined
- Initial draft diagrams or sequence flows are created to illustrate new interactions
- Design decisions and rationale are documented and ready for review

ChangoBuitrago committed:
Listed `v1.0` as an older version.

ChangoBuitrago committed:
Updated title block.

ChangoBuitrago committed:
Updated `Motivation` section.

ChangoBuitrago committed:
Updated Section 2.

ChangoBuitrago committed:
Updated `CRUD Operations` introduction.

ChangoBuitrago committed:
Updated examples to use `"type":"Multikey"` for the verification methods.

ChangoBuitrago committed:
Fixed `@context` lines in the JSON examples.

ChangoBuitrago committed:
Updated `Operations` section.

ChangoBuitrago committed:
Removed `Event Payload` section. Not required anymore.

ChangoBuitrago committed:
Updated `Security Considerations` section.

ChangoBuitrago committed:
Updated  `Privacy Considerations` section.

ChangoBuitrago committed:
Updated  `Reference Implementations and Testing` section.

ChangoBuitrago committed:
Updated `References` section

ChangoBuitrago committed:
Updated `Table of Contents`.

sydor-leocode commented on hedera-did-method-specification.md:
```diff
@@ -80,40 +81,48 @@ Example:
-The method specific identifier `hedera-specific-idstring` is composed of a Hedera network identifier with a `:` separator followed by a `hedera-base58-key` identifier which is a base58 Encoded of a DID root public key and a `did-topic-id` (see details below).
+The method specific identifier (`hedera-specific-idstring` combined with `hedera-specific-parameters`) consists of:
+* A Hedera network identifier (`mainnet` or `testnet`).
+* A Base58 encoded value (`hedera-base58-key`), typically derived from the public key used during the initial creation of the DID. **Crucially, under the v2.0 rules defined in this specification, this `<base58-key>` component serves *only as part of the unique identifier* after creation and *does not* grant ongoing control authority over the DID.**
```
```suggestion
* A Base58 bitcoin encoded value (`hedera-base58btc-key`), typically derived from the public key used during the initial creation of the DID. **Crucially, under the v2.0 rules defined in this specification, this `<base58btc-key>` component serves *only as part of the unique identifier* after creation and *does not* grant ongoing control authority over the DID.**
```

sydor-leocode commented on hedera-did-method-specification.md:
```diff
@@ -80,40 +81,48 @@ Example:
+      "publicKeyMultibase": "z6Mkj1AvU2AEh8ybRqNwHAM3CjbkjYaYHpt9oA1uugW9EVTg6P"
     }
   ],
   "assertionMethod": [
```
```suggestion
  "capabilityInvocation": [
```

sydor-leocode commented on hedera-did-method-specification.md:
```diff
@@ -123,266 +132,207 @@ Example Hedera DID document:
 # 3. CRUD Operations
 
-Create, Update and Delete operations against a DID Document are submitted via a Consensus Service (HCS) message.
+Create, Update, and Deactivate operations against a DID Document under the v2.0 ruleset are performed by submitting authorized messages to the DID's associated Hedera Consensus Service (HCS) topic. Authorization is achieved via cryptographic proofs linked to the DID's designated controller(s), as detailed below.
```
Do we need `create` and `update` at the same time? Since Both holds current DID Document state?

ChangoBuitrago replied:
Yes, we need both. Resolvers need to identify the unique `create` message. It's the anchor point. They use the controller defined in the `create` message's payload to validate the first `update` or `deactivate` operation. If the first message was just labeled "update", the resolver wouldn't know which key/controller is authorized to have submitted it, creating a bootstrapping problem for authorization.

sydor-leocode commented on hedera-did-method-specification.md:
```diff
@@ -123,266 +132,207 @@ Example Hedera DID document:
+- `version`: MUST be the string `"2.0"` to identify the ruleset version being used.
+- `operation`: A string indicating the operation type. Common values include `"create"`, `"update"`, `"deactivate"`. (Note: `"update"` covers modifications to the DID document, potentially including adding or removing properties like verification methods or services, replacing the specific "revoke property" concept from v1.0).
+- `proof`: A mandatory `proof` object whose structure and processing model are based on the **W3C Verifiable Credential Data Integrity v1.0** specification [VC-DI-1.0].
+    - It MUST conform to a specific Data Integrity cryptosuite specification supported by this DID method (e.g., `Ed25519Signature2020`, `JsonWebSignature2020`).
```
```suggestion
    - It MUST conform to a specific Data Integrity cryptosuite specification (e.g., `eddsa-jcs-2022`, `bbs-2023`).
```

sydor-leocode commented on hedera-did-method-specification.md:
```diff
@@ -123,266 +132,207 @@ Example Hedera DID document:
+The security model for Hedera DID Method v2.0 relies on the inherent security of the Hedera network (via HCS) and the robustness of the W3C controller model and cryptographic proofs. Key considerations include:
+
+* **Identifier Component Roles (v2.0 Rule):**
+    * *Crucial Distinction:* The `<base58-key>` component within the DID identifier string (`did:hedera:<network>:<base58-key>_<topic-id>`) serves **only as part of the unique identifier** after the initial creation operation. It **does not grant** ongoing control authority or authorization privileges for managing the DID document under v2.0 rules. Control is solely determined by the `controller` ...
```
```suggestion
    * *Crucial Distinction:* The `<base58btc-key>` component within the DID identifier string (`did:hedera:<network>:<base58btc-key>_<topic-id>`) serves **only as part of the unique identifier** after the initial creation operation. It **does not grant** ongoing control authority or authorization privileges for managing the DID document under v2.0 rules. Control is solely determined by the `controller` property within the DID document and verified via the `proof` mechanism. Misunderstanding this is a security risk.
```

ChangoBuitrago committed:
Removed v1.0 schemas unnecessary for defining how v2.0 operates.

ChangoBuitrago committed:
Added `hedera-did-hcs-message-payload` schema.

ChangoBuitrago committed:
Added `hedera-did-document` schema.

ChangoBuitrago committed:
No Java support.

ChangoBuitrago committed:
No Verifiable Credentials support.

ChangoBuitrago committed:
Update hedera-did-method-specification.md

Co-authored-by: Jakub Sydor <155525776+sydor-leocode@users.noreply.github.com>
Signed-off-by: Pablo Buitrago (w3n:buitrago) <50374632+ChangoBuitrago@users.noreply.github.com>

ChangoBuitrago committed:
Update hedera-did-method-specification.md

Co-authored-by: Jakub Sydor <155525776+sydor-leocode@users.noreply.github.com>
Signed-off-by: Pablo Buitrago (w3n:buitrago) <50374632+ChangoBuitrago@users.noreply.github.com>

ChangoBuitrago committed:
Update hedera-did-method-specification.md

Co-authored-by: Jakub Sydor <155525776+sydor-leocode@users.noreply.github.com>
Signed-off-by: Pablo Buitrago (w3n:buitrago) <50374632+ChangoBuitrago@users.noreply.github.com>

ChangoBuitrago committed:
Update hedera-did-method-specification.md

Co-authored-by: Jakub Sydor <155525776+sydor-leocode@users.noreply.github.com>
Signed-off-by: Pablo Buitrago (w3n:buitrago) <50374632+ChangoBuitrago@users.noreply.github.com>

ChangoBuitrago committed:
Quick fix from `base58` to `base58btc`.

ChangoBuitrago committed:
- Revised JSON examples for improved consistency.
- Amended specification text to align with the definition of capabilityInvocation.

ChangoBuitrago committed:
Updated document readability for clarity.
